cirq/google/op_serializer.py

```diff
@@ -128,6 +137,22 @@ def to_proto(
                 _arg_to_proto(value,
                               out=msg.args[arg.serialized_name],
                               arg_function_language=arg_function_language)
+        if self.auto_serialize_tokens:
+            for tag in op.tags:
+                if isinstance(tag, CalibrationTag):
+                    if constants is not None:
+                        try:
+                            constant = v2.program_pb2.Constant()
+                            constant.string_value = tag.token
+                            msg.constant_index = constants.index(constant)
+                        except ValueError:
+                            # Token not found, add it to the list
+                            msg.constant_index = len(constants)
+                            new_constant = v2.program_pb2.Constant()
+                            new_constant.string_value = tag.token
```

Comment:
Could reuse the same constant object instead of recreating:
```python
constant = v2.program_pb2.Constant()
constant.string_value = tag.token
try:
    msg.constant_index = constants.index(constant)
except ValueError:
    msg.constant_index = len(constants)
    constants.append(constant)
```

Reply:
Done.